cartoframes/data/dataset.py

```diff
@@ -515,20 +517,24 @@ def exists(self):
 
     def is_public(self):
         """Checks to see if table or table used by query has public privacy"""
-        public_context = get_public_context(self.context)
+        public_client = get_client_with_public_creds(self.context)
         try:
-            public_context.sql_client.send('EXPLAIN {}'.format(self.get_query()), do_post=False)
+            public_client.execute_query('EXPLAIN {}'.format(self.get_query()), do_post=False)
             return True
         except CartoException:
             return False
 
+    def _get_client(self):
+        if self._con:
```

Comment:
> Looks like there's some duplication: Do self._con and self.context refer to the same thing?

self._con is the prop and self.context the getter. I agree, I don't like having different names and neither the abbreviation `_con` (seems like DB connection). 

We are going to change `context` by `creds`, but it is out of scope of this PR

> Let's be a little bit more explicit on errors instead of silently failing: I'd add an else and raise an error if there's no such context.

When we are working with local objects (DataFrames or GeoDataFrames) we don't have context and is right. So, we don't have to return an error here

Reply:
Totally forgot the getters thing, yeah it's weird.

We could use always the private attributes inside the classes and let getters for consumers of the API. What do you think?

Reply:
Makes sense